pkgs/sketch_pad/lib/editor/codemirror.dart

```diff
@@ -101,59 +88,52 @@ extension DocExtension on Doc {
   external Position posFromIndex(int index);
 }
 
-@JS()
-@staticInterop
 @anonymous
-class Position {
-  external factory Position({int line, int ch});
-}
-
-extension PositionExtension on Position {
+extension type Position._(JSObject _) implements JSObject {
   external int line;
   external int ch;
-}
 
-@JS()
-@staticInterop
-@anonymous
-class TextMarker {}
+  external factory Position({int line, int ch});
+}
 
-extension TextMarkerExtension on TextMarker {
+extension type TextMarker._(JSObject _) implements JSObject {
   external void clear();
 }
 
-@JS()
-@staticInterop
 @anonymous
-class MarkTextOptions {
+extension type MarkTextOptions._(JSObject _) implements JSObject {
+  external String className;
+  external String title;
+
   external factory MarkTextOptions({
     String className,
     String title,
   });
 }
 
-extension MarkTextOptionsExtension on MarkTextOptions {
-  external String className;
-  external String title;
-}
-
-@JS()
-@staticInterop
 @anonymous
-class HintResults {
-  external factory HintResults({JSArray list, Position from, Position to});
-}
-
-extension HintResultsExtension on HintResults {
-  external JSArray/*<HintResult>*/ list;
+extension type HintResults._(JSObject _) implements JSObject {
```

Comment:
What does `._(JSObject _)` do?

Reply:
`._` makes the primary constructor private so it can't be used outside this library and `(JSObject _)` means the extension type's representation object/field (what it wraps) is a `JSObject`. I gave it the name `_` since we don't need to access it within the declaration.

Reply:
👍